### ISSUE TITLE
How to current speed limit of the road particularly for the Truck?

### ISSUE THREAD

**Issue by binod-techindustan** [heremaps/here-android-sdk-examples#351]
Now currently we are calculating speed limit using. Please check wether I am implementing it right or not.
```
private PositioningManager.OnPositionChangedListener mapPositionHandler = new PositioningManager.OnPositionChangedListener() {
        @Override
        public void onPositionUpdated(PositioningManager.LocationMethod method, GeoPosition position,
                                      boolean isMapMatched) {
            Log.e("PostionChanged", "Called");
       if (position.isValid() && position instanceof MatchedGeoPosition) {
            MatchedGeoPosition mgp = (MatchedGeoPosition) geoPosition;
            float currentSpeedLimit=mgp.getRoadElement().getSpeedLimit();
            }

        }

        @Override
        public void onPositionFixChanged(PositioningManager.LocationMethod method,
                                         PositioningManager.LocationStatus status) {
     

        }
    };

```

**Comment by NazarKacharaba** [comment]
Hi @binod-techindustan 
That is correct. Note, that you must use `TransportMode.TRUCK` for route calculation, and then start navigation with that route, then `RoadElement` will have truck's speed limit.

Also, you can use `NavigationManager.SpeedWarningListener` API to notify user that speed was exceeded or otherwise.

**Comment by binod-techindustan** [comment]
Yes @NazarKacharaba I am setting `TransportMode.TRUCK` while creating route.

**Comment by binod-techindustan** [comment]
@NazarKacharaba the issue is freeways speed limit in US for truck is 55 mph in most case but we are getting 65 mph. The values in android is not similar in iOS app too.

**Comment by NazarKacharaba** [comment]
@binod-techindustan 
Could you provide coordinates for routes?

**Comment by binod-techindustan** [comment]
@NazarKacharaba one our user report it. It was in Highway no 80 California

**Comment by NazarKacharaba** [comment]
@binod-techindustan We are looking into this.

**Comment by VladChernetsky** [comment]
Hi @NazarKacharaba 
I've faced the same issue. I get car speed limits with applied TransportMode.TRUCK.
Route waypoints: Lithuania Kaunas - Lithuania Klaipeda

**Comment by NazarKacharaba** [comment]
@binod-techindustan Speed values on  Highway no 80 California, USA is fixed for truck in the latest map update. Please use `MapLoader.performMapDataUpdate() ` to switch to the latest map version.

@VladChernetsky we will take a look at this routes

**Comment by binod-techindustan** [comment]
@NazarKacharaba where can I call this method. Should I call this method before loading map? Will there be an issue if I we call during map initialisation. So let me know the best place to perform the map updates.

**Comment by NazarKacharaba** [comment]
You can find some instructions [here](https://developer.here.com/documentation/android-premium/3.15/dev_guide/topics/maps-offline-maploader.html)

**Comment by NazarKacharaba** [comment]
You should perform map data update after you initialized `MapEngine`, before any route calculation or navigation. Check if new update is available using `checkForMapDataUpdate ` method, and then update map calling `performMapDataUpdate `.
